enclave-runtime/src/test/sidechain_aura_tests.rs

```diff
@@ -0,0 +1,239 @@
+		exec_aura_on_slot::<_, ParentchainBlock, SignedSidechainBlock, _, _>(
+			slot_info,
+			signer,
+			ocall_api.as_ref().clone(),
+			proposer_environment,
+			shards,
+		)
+		.unwrap();
+
+	assert_eq!(1, blocks.len());
+	assert_eq!(
+		state_hash_before_block_production,
+		get_state_hash(state_handler.as_ref(), &shard_id)
+	);
+
+	// Ensure that invalid calls are removed from pool. Valid calls should only be removed upon block import.
+	assert_eq!(1, rpc_author.get_pending_tops_separated(shard_id).unwrap().0.len());
+
+	info!("Executed AURA successfully. Sending blocks and extrinsics..");
+	let propose_to_block_import_ocall_api =
+		ProposeToImportOCallApi::new(parentchain_header, block_importer);
+
+	send_blocks_and_extrinsics::<ParentchainBlock, _, _, _, _>(
+		blocks,
+		opaque_calls,
+		propose_to_block_import_ocall_api,
+		&validator_access,
+		&extrinsics_factory,
+	)
+	.unwrap();
+
+	// After importing the sidechain block, the trusted operation should be removed.
+	assert!(rpc_author.get_pending_tops_separated(shard_id).unwrap().0.is_empty());
+
+	// After importing the block, the state hash must be changed.
+	assert_ne!(
+		state_hash_before_block_production,
+		get_state_hash(state_handler.as_ref(), &shard_id)
+	);
+	let mut state = state_handler.load_initialized(&shard_id).unwrap();
+	let free_balance = Stf::account_data(&mut state, &receiver.public().into()).unwrap().free;
+	assert_eq!(free_balance, transfered_amount);
+}
+
+fn encrypted_trusted_operation_transfer_balance<
+	AttestationApi: EnclaveAttestationOCallApi,
+	ShieldingKey: ShieldingCrypto,
+>(
+	attestation_api: &AttestationApi,
+	shard_id: &ShardIdentifier,
+	shielding_key: &ShieldingKey,
+	from: ed25519::Pair,
+	to: AccountId,
+	amount: Balance,
+) -> Vec<u8> {
+	let mr_enclave = attestation_api.get_mrenclave_of_self().unwrap();
+
+	let call = TrustedCall::balance_transfer(from.public().into(), to, amount).sign(
+		&KeyPair::Ed25519(from),
+		0,
+		&mr_enclave.m,
+		shard_id,
+	);
+
+	let trusted_operation = TrustedOperation::direct_call(call);
+	let encoded_operation = trusted_operation.encode();
+
+	shielding_key.encrypt(encoded_operation.as_slice()).unwrap()
+}
+
+fn get_state_hash(state_handler: &HandleStateMock, shard_id: &ShardIdentifier) -> H256 {
+	let state = state_handler.load_initialized(shard_id).unwrap();
+	let sidechain_state = TestSidechainDb::new(state);
+	sidechain_state.state_hash()
+}
+
+fn create_top_pool() -> Arc<TestTopPool> {
+	let rpc_responder = Arc::new(TestRpcResponder::new());
+	let sidechain_api = Arc::new(SidechainApi::<ParentchainBlock>::new());
+	Arc::new(TestTopPool::create(PoolOptions::default(), sidechain_api, rpc_responder))
+}
+
+fn create_ocall_api(signer: &TestSigner) -> Arc<TestOCallApi> {
+	let enclave_validateer = Enclave::new(
+		signer.public().into(),
+		Default::default(),
+		Default::default(),
+		Default::default(),
+	);
+	Arc::new(TestOCallApi::default().with_validateer_set(Some(vec![enclave_validateer])))
+}
```

Comment:
Nice extensive test! 👍

Reply:
It's not ideal TBH, it will have to be touched and modified a lot as we work on the code, which is not great. But for this PR, it's what I could come up with without blowing the PR even more out of proportion.